packages/manager/apps/cloud/client/app/dbaas/logs/detail/kibana/add/logs-kibana-add.controller.js

```diff
@@ -0,0 +1,142 @@
+class LogsKibanaAddModalCtrl {
+  constructor(
+    $q,
+    $stateParams,
+    $uibModalInstance,
+    $translate,
+    CucControllerHelper,
+    kibanaInfo,
+    LogsKibanaService,
+    LogsConstants,
+  ) {
+    this.$stateParams = $stateParams;
+    this.$q = $q;
+    this.CucControllerHelper = CucControllerHelper;
+    this.kibanaInfo = kibanaInfo;
+    this.LogsKibanaService = LogsKibanaService;
+    this.LogsConstants = LogsConstants;
+    this.$uibModalInstance = $uibModalInstance;
+    this.$translate = $translate;
+    this.serviceName = $stateParams.serviceName;
+    this.kibana = this.LogsKibanaService.getNewKibana();
+    this.kibanaPrice = { price: '' };
+    this.initLoaders();
+  }
+
+  $onInit() {
+    this.isEdit = this.constructor.checkIsEdit(this.kibanaInfo);
+    if (this.isEdit) {
+      this.populateKibana();
+    } else {
+      this.clearKibana();
```

Comment:
Can you move else part into a separate method? this will improve readability of the code

Reply:
Done.